### ISSUE TITLE
Duplication Bug

### ISSUE THREAD

**Issue by PSYCHO1512** [Hebbinkpro/MagicCrates#31]
how can we add potions to the crates

**Comment by Hebbinkpro** [comment]
Hi, you can add potions the same way as other items.
For the item ID, you can look at the [potions item data](https://minecraft.wiki/w/Potion#Item_data) on the Minecraft wiki, and put `minecraft:<PotionType>_potion` in your `crate_types.json`

For example, if you want to give the strength potion, you can use `minecraft:strength_potion`, `minecraft:strong_strength_potion` or `minecraft:long_strength_potion`.

Keep in mind that not all potions listed on the Minecraft wiki are available in PocketMine.

**Comment by PSYCHO1512** [comment]
thansk for that and can we change the probabilty of the items

**Comment by Hebbinkpro** [comment]
Yes, it is possible to change the probability of items.
Take for example the default common crate:
```
  "common": {
    "name": "§cCommon §6Crate",
    "rewards": {
      "iron": {
        "name": "Common Iron",
        "amount": 5,
        "item": {
          "id": "iron_ingot"
        }
      },
      "dirt": {
        "name": "Dirt",
        "amount": 15,
        "item": {
          "id": "dirt",
          "amount": 2
        }
      }
    }
  },
```
The reward iron has an amount of 5, and the dirt reward has an amount of 15.
This means that there are a total of 20 items, of which 5 are iron and 15 are dirt, resulting in a probability of 25% for Iron and 75% for Dirt.

So, by changing the amounts you can customize the probabilities of your items.

For more information about how to use the plugin have a look at the [documentation](https://github.com/Hebbinkpro/MagicCrates?tab=readme-ov-file#crate-types-1).

**Comment by PSYCHO1512** [comment]
one more thing could you look at the crates i create they only give one item and it dupes iteslf
{
  "common": {
    "name": "§7Common §6Crate",
    "rewards": {
      "iron_ingot": {
        "name": "Iron_ingot",
        "amount": 1,
        "item": {
          "id": "iron_ingot"
        }
      },
      "gold_ingot": {
        "name": "Gold_ingot",
        "amount": 1,
        "item": {
          "id": "gold_ingot"
        }
      },
      "lapis_lazuli": {
        "name": "lapis_lazuli",
        "amount": 1,
        "item": {
          "id": "lapis_lazuli"
        }
      },
      "diamond": {
        "name": "Diamond",
        "amount": 1,
        "item": {
          "id": "diamond"
        }
      },
      "emerald": {
        "name": "Emerald",
        "amount": 1,
        "item": {
          "id": "emerald"
        }
      },
      "golden_apple": {
        "name": "Golden apple",
        "amount": 3,
        "item": {
          "id": "golden_apple"
        }
      }
    }
  },
  "rare": {
    "name": "§aRare §6Crate",
    "rewards": {
      "iron_ingot": {
        "name": "Iron_ingot",
        "amount": 3,
        "item": {
          "id": "iron_ingot"
        }
      },
      "gold_ingot": {
        "name": "Gold_ingot",
        "amount": 3,
        "item": {
          "id": "gold_ingot"
        }
      },
      "lapis_lazuli": {
        "name": "lapis_lazuli",
        "amount": 3,
        "item": {
          "id": "lapis_lazuli"
        }
      },
      "diamond": {
        "name": "Diamond",
        "amount": 3,
        "item": {
          "id": "diamond"
        }
      },
      "emerald": {
        "name": "Emerald",
        "amount": 3,
        "item": {
          "id": "emerald"
        }
      },
      "golden_apple": {
        "name": "Golden apple",
        "amount": 3,
        "item": {
          "id": "golden_apple"
        }
      },
      "obsidian": {
        "name": "Obsidian",
        "amount": 3,
        "item": {
          "id": "obsidian"
        }
      },
      "tnt": {
        "name": "TnT",
        "amount": 3,
        "item": {
          "id": "tnt"
        }
      }
    }
  },
  "epic": {
    "name": "§bEpic §6Crate",
    "rewards": {
      "iron_block": {
        "name": "Iron_block",
        "amount": 2,
        "item": {
          "id": "iron_block"
        }
      },
      "gold_block": {
        "name": "Gold_block",
        "amount": 2,
        "item": {
          "id": "gold_block"
        }
      },
      "lapis_block": {
        "name": "lapis_block",
        "amount": 2,
        "item": {
          "id": "lapis_block"
        }
      },
      "diamond_block": {
        "name": "Diamond Block",
        "amount": 1,
        "item": {
          "id": "diamond_block"
        }
      },
      "emerald_block": {
        "name": "Emerald Block",
        "amount": 1,
        "item": {
          "id": "emerald_block"
        }
      },
      "golden_apple": {
        "name": "Golden apple",
        "amount": 10,
        "item": {
          "id": "golden_apple"
        }
      },
      "obsidian": {
        "name": "Obsidian",
        "amount": 10,
        "item": {
          "id": "obsidian"
        }
      },
      "tnt": {
        "name": "TnT",
        "amount": 10,
        "item": {
          "id": "tnt"
        }
      }
    }
  },
  "legendary": {
    "name": "§eLegendary §6Crate",
    "rewards": {
      "iron_block": {
        "name": "Iron block",
        "amount": 5,
        "item": {
          "id": "iron_block"
        }
      },
      "gold_block": {
        "name": "Gold block",
        "amount": 5,
        "item": {
          "id": "gold_block"
        }
      },
      "lapis_block": {
        "name": "lapis block",
        "amount": 5,
        "item": {
          "id": "lapis_block"
        }
      },
      "diamond_block": {
        "name": "Diamond Block",
        "amount": 5,
        "item": {
          "id": "diamond_block"
        }
      },
      "emerald_block": {
        "name": "Emerald Block",
        "amount": 5,
        "item": {
          "id": "emerald_block"
        }
      },
      "golden_apple": {
        "name": "Golden apple",
        "amount": 16,
        "item": {
          "id": "golden_apple"
        }
      },
      "obsidian": {
        "name": "Obsidian",
        "amount": 16,
        "item": {
          "id": "obsidian"
        }
      },
      "tnt": {
        "name": "TnT",
        "amount": 20,
        "item": {
          "id": "tnt"
        }
      },
      "paper": {
        "name": "paper",
        "amount": 1,
        "item": {
          "id": "paper",
          "commands": [
            "give {player} 52:12 1"
          ]
        }
      },
      "enchanted_golden_apple": {
        "name": "Enchanted Golden apple",
        "amount": 9,
        "item": {
          "id": "enchanted_golden_apple"
        }
      },
      "cow_spawner": {
        "name": "Cow Spawner",
        "amount": 1,
        "item": {
          "id": "paper",
          "commands": [
            "give {player} 52:11 1"
          ]
        }
      },
      "zombie_spawner": {
        "name": "Zombie Spawner",
        "amount": 1,
        "item": {
          "id": "paper",
          "commands": [
            "give {player} 52:32 1"
          ]
        }
      },
      "skeleton_spawner": {
        "name": "Skeleton Spawner",
        "amount": 1,
        "item": {
          "id": "paper",
          "commands": [
            "give {player} 52:34 1"
          ]
        }
      },
      "spider_spawner": {
        "name": "Spider Spawner",
        "amount": 1,
        "item": {
          "id": "paper",
          "commands": [
            "give {player} 52:35 1"
          ]
        }
      },
      "pig_spawner": {
        "name": "Pig Spawner",
        "amount": 1,
        "item": {
          "id": "paper",
          "commands": [
            "give {player} 52:12 1"
          ]
        }
      },
      "blaze_spawner": {
        "name": "Blaze Spawner",
        "amount": 1,
        "item": {
          "id": "paper",
          "commands": [
            "give {player} 52:43 1"
          ]
        }
      }
    }
  },
  "nebula": {
    "name": "§5Nebula §6Crate",
    "rewards": {
      "diamond_block": {
        "name": "Diamond Block",
        "amount": 9,
        "item": {
          "id": "diamond_block"
        }
      },
      "emerald_block": {
        "name": "Emerald Block",
        "amount": 9,
        "item": {
          "id": "emerald_block"
        }
      },
      "enchanted_golden_apple": {
        "name": "Enchanted Golden apple",
        "amount": 9,
        "item": {
          "id": "enchanted_golden_apple"
        }
      },
      "golden_apple": {
        "name": "Golden apple",
        "amount": 20,
        "item": {
          "id": "golden_apple"
        }
      },
      "iron_golem_spawner": {
        "name": "Iron Golem Spawner",
        "amount": 1,
        "item": {
          "id": "paper",
          "commands": [
            "give {player} 52:20 1"
          ]
        }
      },
      "blaze_spawner": {
        "name": "Blaze Spawner",
        "amount": 1,
        "item": {
          "id": "paper",
          "commands": [
            "give {player} 52:43 1"
          ]
        }
      },
      "creeper_spawner": {
        "name": "Creeper Spawner",
        "amount": 1,
        "item": {
          "id": "paper",
          "commands": [
            "give {player} 52:33 1"
          ]
        }
      },
      "nebula_helmet": {
        "name": "§dNebula Helmet",
        "amount": 1,
        "item": {
          "id": "diamond_helmet",
          "name": "§dNebula Helmet",
          "enchantments": [
            {
              "name": "protection",
              "level": 4
            },
            {
              "name": "unbreaking",
              "level": 3
            }
          ]
        }
      },
      "nebula_chestplate": {
        "name": "§dNebula Chestplate",
        "amount": 1,
        "item": {
          "id": "diamond_chestplate",
          "name": "§dNebula Chestplate",
          "enchantments": [
            {
              "name": "protection",
              "level": 4
            },
            {
              "name": "unbreaking",
              "level": 3
            }
          ]
        }
      },
      "nebula_boots": {
        "name": "§dNebula Boots",
        "amount": 1,
        "item": {
          "id": "diamond_boots",
          "name": "§dNebula Boots",
          "enchantments": [
            {
              "name": "protection",
              "level": 4
            },
            {
              "name": "unbreaking",
              "level": 3
            }
          ]
        }
      },
      "nebula_leggings": {
        "name": "§dNebula Leggings",
        "amount": 1,
        "item": {
          "id": "diamond_leggings",
          "name": "§dNebula Leggings",
          "enchantments": [
            {
              "name": "protection",
              "level": 4
            },
            {
              "name": "unbreaking",
              "level": 3
            }
          ]
        }
      },
      "nebula_sword": {
        "name": "§dNebula Sword",
        "amount": 1,
        "item": {
          "id": "diamond_sword",
          "name": "§dNebula Sword",
          "enchantments": [
            {
              "name": "sharpness",
              "level": 5
            },
            {
              "name": "unbreaking",
              "level": 3
            },
            {
              "name": "fire_aspect",
              "level": 2
            }
          ]
        }
      }
    }
  },
  "koth": {
    "name": "§4KOTH §6Crate",
    "rewards": {
      "koth_helmet": {
        "name": "§4KOTH Helmet",
        "amount": 1,
        "item": {
          "id": "diamond_helmet",
          "name": "§4KOTH Helmet",
          "enchantments": [
            {
              "name": "protection",
              "level": 5
            },
            {
              "name": "unbreaking",
              "level": 4
            }
          ]
        }
      },
      "koth_chestplate": {
        "name": "§4KOTH Chestplate",
        "amount": 1,
        "item": {
          "id": "diamond_chestplate",
          "name": "§4KOTH Chestplate",
          "enchantments": [
            {
              "name": "protection",
              "level": 5
            },
            {
              "name": "unbreaking",
              "level": 4
            }
          ]
        }
      },
      "koth_boots": {
        "name": "§4KOTH Boots",
        "amount": 1,
        "item": {
          "id": "diamond_boots",
          "name": "§4KOTH Boots",
          "enchantments": [
            {
              "name": "protection",
              "level": 5
            },
            {
              "name": "unbreaking",
              "level": 4
            }
          ]
        }
      },
      "koth_leggings": {
        "name": "§4KOTH Leggings",
        "amount": 1,
        "item": {
          "id": "diamond_leggings",
          "name": "§4KOTH Leggings",
          "enchantments": [
            {
              "name": "protection",
              "level": 5
            },
            {
              "name": "unbreaking",
              "level": 4
            }
          ]
        }
      },
      "koth_sword": {
        "name": "§4KOTH Sword",
        "amount": 1,
        "item": {
          "id": "diamond_sword",
          "name": "§4KOTH Sword",
          "enchantments": [
            {
              "name": "sharpness",
              "level": 6
            },
            {
              "name": "unbreaking",
              "level": 4
            },
            {
              "name": "fire_aspect",
              "level": 3
            }
          ]
        }
      },
      "koth_pickaxe": {
        "name": "§4KOTH Pickaxe",
        "amount": 1,
        "item": {
          "id": "diamond_pickaxe",
          "name": "§4KOTH Pickaxe",
          "enchantments": [
            {
              "name": "efficiency",
              "level": 6
            },
            {
              "name": "unbreaking",
              "level": 5
            },
            {
              "name": "fortune",
              "level": 4
            }
          ]
        }
      }
    }
  }
}

**Comment by PSYCHO1512** [comment]
> one more thing could you look at the crates i create they only give one item and it dupes iteslf { "common": { "name": "§7Common §6Crate", "rewards": { "iron_ingot": { "name": "Iron_ingot", "amount": 1, "item": { "id": "iron_ingot" } }, "gold_ingot": { "name": "Gold_ingot", "amount": 1, "item": { "id": "gold_ingot" } }, "lapis_lazuli": { "name": "lapis_lazuli", "amount": 1, "item": { "id": "lapis_lazuli" } }, "diamond": { "name": "Diamond", "amount": 1, "item": { "id": "diamond" } }, "emerald": { "name": "Emerald", "amount": 1, "item": { "id": "emerald" } }, "golden_apple": { "name": "Golden apple", "amount": 3, "item": { "id": "golden_apple" } } } }, "rare": { "name": "§aRare §6Crate", "rewards": { "iron_ingot": { "name": "Iron_ingot", "amount": 3, "item": { "id": "iron_ingot" } }, "gold_ingot": { "name": "Gold_ingot", "amount": 3, "item": { "id": "gold_ingot" } }, "lapis_lazuli": { "name": "lapis_lazuli", "amount": 3, "item": { "id": "lapis_lazuli" } }, "diamond": { "name": "Diamond", "amount": 3, "item": { "id": "diamond" } }, "emerald": { "name": "Emerald", "amount": 3, "item": { "id": "emerald" } }, "golden_apple": { "name": "Golden apple", "amount": 3, "item": { "id": "golden_apple" } }, "obsidian": { "name": "Obsidian", "amount": 3, "item": { "id": "obsidian" } }, "tnt": { "name": "TnT", "amount": 3, "item": { "id": "tnt" } } } }, "epic": { "name": "§bEpic §6Crate", "rewards": { "iron_block": { "name": "Iron_block", "amount": 2, "item": { "id": "iron_block" } }, "gold_block": { "name": "Gold_block", "amount": 2, "item": { "id": "gold_block" } }, "lapis_block": { "name": "lapis_block", "amount": 2, "item": { "id": "lapis_block" } }, "diamond_block": { "name": "Diamond Block", "amount": 1, "item": { "id": "diamond_block" } }, "emerald_block": { "name": "Emerald Block", "amount": 1, "item": { "id": "emerald_block" } }, "golden_apple": { "name": "Golden apple", "amount": 10, "item": { "id": "golden_apple" } }, "obsidian": { "name": "Obsidian", "amount": 10, "item": { "id": "obsidian" } }, "tnt": { "name": "TnT", "amount": 10, "item": { "id": "tnt" } } } }, "legendary": { "name": "§eLegendary §6Crate", "rewards": { "iron_block": { "name": "Iron block", "amount": 5, "item": { "id": "iron_block" } }, "gold_block": { "name": "Gold block", "amount": 5, "item": { "id": "gold_block" } }, "lapis_block": { "name": "lapis block", "amount": 5, "item": { "id": "lapis_block" } }, "diamond_block": { "name": "Diamond Block", "amount": 5, "item": { "id": "diamond_block" } }, "emerald_block": { "name": "Emerald Block", "amount": 5, "item": { "id": "emerald_block" } }, "golden_apple": { "name": "Golden apple", "amount": 16, "item": { "id": "golden_apple" } }, "obsidian": { "name": "Obsidian", "amount": 16, "item": { "id": "obsidian" } }, "tnt": { "name": "TnT", "amount": 20, "item": { "id": "tnt" } }, "paper": { "name": "paper", "amount": 1, "item": { "id": "paper", "commands": [ "give {player} 52:12 1" ] } }, "enchanted_golden_apple": { "name": "Enchanted Golden apple", "amount": 9, "item": { "id": "enchanted_golden_apple" } }, "cow_spawner": { "name": "Cow Spawner", "amount": 1, "item": { "id": "paper", "commands": [ "give {player} 52:11 1" ] } }, "zombie_spawner": { "name": "Zombie Spawner", "amount": 1, "item": { "id": "paper", "commands": [ "give {player} 52:32 1" ] } }, "skeleton_spawner": { "name": "Skeleton Spawner", "amount": 1, "item": { "id": "paper", "commands": [ "give {player} 52:34 1" ] } }, "spider_spawner": { "name": "Spider Spawner", "amount": 1, "item": { "id": "paper", "commands": [ "give {player} 52:35 1" ] } }, "pig_spawner": { "name": "Pig Spawner", "amount": 1, "item": { "id": "paper", "commands": [ "give {player} 52:12 1" ] } }, "blaze_spawner": { "name": "Blaze Spawner", "amount": 1, "item": { "id": "paper", "commands": [ "give {player} 52:43 1" ] } } } }, "nebula": { "name": "§5Nebula §6Crate", "rewards": { "diamond_block": { "name": "Diamond Block", "amount": 9, "item": { "id": "diamond_block" } }, "emerald_block": { "name": "Emerald Block", "amount": 9, "item": { "id": "emerald_block" } }, "enchanted_golden_apple": { "name": "Enchanted Golden apple", "amount": 9, "item": { "id": "enchanted_golden_apple" } }, "golden_apple": { "name": "Golden apple", "amount": 20, "item": { "id": "golden_apple" } }, "iron_golem_spawner": { "name": "Iron Golem Spawner", "amount": 1, "item": { "id": "paper", "commands": [ "give {player} 52:20 1" ] } }, "blaze_spawner": { "name": "Blaze Spawner", "amount": 1, "item": { "id": "paper", "commands": [ "give {player} 52:43 1" ] } }, "creeper_spawner": { "name": "Creeper Spawner", "amount": 1, "item": { "id": "paper", "commands": [ "give {player} 52:33 1" ] } }, "nebula_helmet": { "name": "§dNebula Helmet", "amount": 1, "item": { "id": "diamond_helmet", "name": "§dNebula Helmet", "enchantments": [ { "name": "protection", "level": 4 }, { "name": "unbreaking", "level": 3 } ] } }, "nebula_chestplate": { "name": "§dNebula Chestplate", "amount": 1, "item": { "id": "diamond_chestplate", "name": "§dNebula Chestplate", "enchantments": [ { "name": "protection", "level": 4 }, { "name": "unbreaking", "level": 3 } ] } }, "nebula_boots": { "name": "§dNebula Boots", "amount": 1, "item": { "id": "diamond_boots", "name": "§dNebula Boots", "enchantments": [ { "name": "protection", "level": 4 }, { "name": "unbreaking", "level": 3 } ] } }, "nebula_leggings": { "name": "§dNebula Leggings", "amount": 1, "item": { "id": "diamond_leggings", "name": "§dNebula Leggings", "enchantments": [ { "name": "protection", "level": 4 }, { "name": "unbreaking", "level": 3 } ] } }, "nebula_sword": { "name": "§dNebula Sword", "amount": 1, "item": { "id": "diamond_sword", "name": "§dNebula Sword", "enchantments": [ { "name": "sharpness", "level": 5 }, { "name": "unbreaking", "level": 3 }, { "name": "fire_aspect", "level": 2 } ] } } } }, "koth": { "name": "§4KOTH §6Crate", "rewards": { "koth_helmet": { "name": "§4KOTH Helmet", "amount": 1, "item": { "id": "diamond_helmet", "name": "§4KOTH Helmet", "enchantments": [ { "name": "protection", "level": 5 }, { "name": "unbreaking", "level": 4 } ] } }, "koth_chestplate": { "name": "§4KOTH Chestplate", "amount": 1, "item": { "id": "diamond_chestplate", "name": "§4KOTH Chestplate", "enchantments": [ { "name": "protection", "level": 5 }, { "name": "unbreaking", "level": 4 } ] } }, "koth_boots": { "name": "§4KOTH Boots", "amount": 1, "item": { "id": "diamond_boots", "name": "§4KOTH Boots", "enchantments": [ { "name": "protection", "level": 5 }, { "name": "unbreaking", "level": 4 } ] } }, "koth_leggings": { "name": "§4KOTH Leggings", "amount": 1, "item": { "id": "diamond_leggings", "name": "§4KOTH Leggings", "enchantments": [ { "name": "protection", "level": 5 }, { "name": "unbreaking", "level": 4 } ] } }, "koth_sword": { "name": "§4KOTH Sword", "amount": 1, "item": { "id": "diamond_sword", "name": "§4KOTH Sword", "enchantments": [ { "name": "sharpness", "level": 6 }, { "name": "unbreaking", "level": 4 }, { "name": "fire_aspect", "level": 3 } ] } }, "koth_pickaxe": { "name": "§4KOTH Pickaxe", "amount": 1, "item": { "id": "diamond_pickaxe", "name": "§4KOTH Pickaxe", "enchantments": [ { "name": "efficiency", "level": 6 }, { "name": "unbreaking", "level": 5 }, { "name": "fortune", "level": 4 } ] } } } } }

like theres 3 golden apples in the common create but i it only give me 2 and even when we get diamonds its only supposed to give 1 diamond but it gives 2 same with armour and other stuff

**Comment by Hebbinkpro** [comment]
I'm not sure how the bug occurs, but it is possible this is due to one of the recent pmmp updates, I have to look into this further which will take some time.

All crates only give one item because you have not specified the `amount` of the item, but you only set the `amount` which corresponds to the probability of getting the reward, as can be seen in the [documentation](https://github.com/Hebbinkpro/MagicCrates?tab=readme-ov-file#item).

This is your common crate
``` 
    "common": {
        "name": "§7Common §6Crate",
        "rewards": {
            "iron_ingot": {
                "name": "Iron_ingot",
                "amount": 1,
                "item": {
                    "id": "iron_ingot"
                }
            },
            "gold_ingot": {
                "name": "Gold_ingot",
                "amount": 1,
                "item": {
                    "id": "gold_ingot"
                }
            },
            "lapis_lazuli": {
                "name": "lapis_lazuli",
                "amount": 1,
                "item": {
                    "id": "lapis_lazuli"
                }
            },
            "diamond": {
                "name": "Diamond",
                "amount": 1,
                "item": {
                    "id": "diamond"
                }
            },
            "emerald": {
                "name": "Emerald",
                "amount": 1,
                "item": {
                    "id": "emerald"
                }
            },
            "golden_apple": {
                "name": "Golden apple",
                "amount": 3,
                "item": {
                    "id": "golden_apple"
                }
            }
        }
    },
``` 
For example, I assume that when you win the "golden_apple" **reward**, you want to receive 3 golden apple **items**.
Currently, you will only receive one, since the item only contains an ID.
If you add amount to the item, then you can get more items.
```
           "golden_apple": {
                "name": "Golden apple",
                "amount": 3,
                "item": {
                    "id": "golden_apple"
                    "amount": 3
                }
            }
``` 
This will give you 3 golden apples when you win the golden apple reward.

The amount given in the reward (not inside item), determines the probability of receiving that specific reward. In the case of the golden apple is this 3, and the total of items in the crate is 8 (sum of amounts of all rewards in the crate), so the probability of getting the golden apple reward is 3/8=0.375 (or 37.5%).

**Comment by PSYCHO1512** [comment]
i did that but its still give 2 of any items

**Comment by PSYCHO1512** [comment]
and the yml that the plugin used to use was better than this json file format in my opinion

**Comment by PSYCHO1512** [comment]
whats your discord i can send you like a video of whats happening

**Comment by Hebbinkpro** [comment]
I'm not sure how the bug occurs, but it is possible this is due to one of the recent pmmp updates, I have to look into this further which will take some time.

All crates only give one item because you have not specified the `amount` of the item, but you only set the `amount` which corresponds to the probability of getting the reward, as can be seen in the [documentation](https://github.com/Hebbinkpro/MagicCrates?tab=readme-ov-file#item).

This is your common crate
``` 
    "common": {
        "name": "§7Common §6Crate",
        "rewards": {
            "iron_ingot": {
                "name": "Iron_ingot",
                "amount": 1,
                "item": {
                    "id": "iron_ingot"
                }
            },
            "gold_ingot": {
                "name": "Gold_ingot",
                "amount": 1,
                "item": {
                    "id": "gold_ingot"
                }
            },
            "lapis_lazuli": {
                "name": "lapis_lazuli",
                "amount": 1,
                "item": {
                    "id": "lapis_lazuli"
                }
            },
            "diamond": {
                "name": "Diamond",
                "amount": 1,
                "item": {
                    "id": "diamond"
                }
            },
            "emerald": {
                "name": "Emerald",
                "amount": 1,
                "item": {
                    "id": "emerald"
                }
            },
            "golden_apple": {
                "name": "Golden apple",
                "amount": 3,
                "item": {
                    "id": "golden_apple"
                }
            }
        }
    },
``` 
For example, I assume that when you win the "golden_apple" **reward**, you want to receive 3 golden apple **items**.
Currently, you will only receive one, since the item only contains an ID.
If you add amount to the item, then you can get more items.
```
           "golden_apple": {
                "name": "Golden apple",
                "amount": 3,
                "item": {
                    "id": "golden_apple"
                    "amount": 3
                }
            }
``` 
This will give you 3 golden apples when you win the golden apple reward.

The amount given in the reward (not inside item), determines the probability of receiving that specific reward. In the case of the golden apple is this 3, and the total of items in the crate is 8 (sum of amounts of all rewards in the crate), so the probability of getting the golden apple reward is 3/8=0.375 (or 37.5%).

**Comment by Hebbinkpro** [comment]
The change from yml to JSON has nothing to do with the duplication bug, and is only a different notation.

As said before, I will look into the duplication bug as this is probably caused by a recent PMMP update, and this will take some time.

If you want you can try upload the video here on GitHub.

**Comment by Hebbinkpro** [comment]
The duplication issue has been fixed in v3.1.1. You can download the latest phar [here](https://poggit.pmmp.io/ci/Hebbinkpro/MagicCrates/MagicCrates)
